### PR TITLE
Vmod proxy

### PR DESCRIPTION
**Pull request by ehocdet**
Introduce vmod_proxy to extract TLV attributes from proxy-protocol-v2.
Is related to #2561

### CODE REVIEW

**Comment by bsdphk** [comment]
A bit of live experience would be nice.

**Comment by ehocdet** [comment]
The first use is to report the cryptographic information to the backends finally to qualify the security level of the TLS connection.
HAproxy can generate SSL TLV and 1.9dev (http://git.haproxy.org/?p=haproxy.git) have all SSL TLV attributes.

File: bin/varnishtest/tests/o00005.vtc [hunk 1]
```diff
@@ -0,0 +1,41 @@
+varnishtest "PROXY v2 TLV test"
+
+server s1 {
+	rxreq
+	txresp
+} -start
+
+varnish v1 -proto "PROXY" -vcl+backend {
+	import proxy from "${topbuild}/lib/libvmod_proxy/.libs/";
```
Comment (nigoroll) [inline bin/varnishtest/tests/o00005.vtc]: from ... not needed any more

File: bin/varnishd/proxy/cache_proxy_proto.c [hunk 2]
```diff
@@ -423,6 +429,15 @@ vpx_proto2(const struct worker *wrk, struct req *req)
 		VSL(SLT_ProxyGarbage, req->sp->vxid, "PROXY2: header length mismatch");
 		return (0);
 	}
+	if (tlv_len) {
+		SES_Reserve_xport_priv(req->sp, &up);
+		WS_Reserve(req->sp->ws, 2 + tlv_len);
```
Comment (nigoroll) [inline bin/varnishd/proxy/cache_proxy_proto.c]: check for error

File: bin/varnishtest/tests/o00005.vtc (hunk 1 above)
Comment (Dridi) [reply bin/varnishtest/tests/o00005.vtc]: in order to drop the from clause `bin/varnishtest/vmods.h` needs to be updated.

Comment (ehocdet) [reply bin/varnishtest/tests/o00005.vtc]: fixed and rebase, thanks.

File: bin/varnishd/proxy/cache_proxy_proto.c (hunk 2 above)
Comment (ehocdet) [reply bin/varnishd/proxy/cache_proxy_proto.c]: fixed, also remove WS_MarkOverflow in vmod_proxy.c (already set in WS_Reserve)

**Comment by ehocdet** [comment]
rebase and refer man and doc.

**Pull request merged**